### ISSUE TITLE
Switching Locations On/Off

### ISSUE THREAD

**Issue by Jacki3** [LoGaCulture/LUTE#35]
There will be times that games may need to be shut off or specific locations may need to be turned off in a specific experience.

**Describe the solution you'd like**
We should offer a way to shut specific locations off and then orders/nodes that relate to that location will either shut off or use a backup location that derives from the location variable. These methods need to be highly accesible and robust to work with an api hook that will be managed by server-side functions.

**Additional context**
Start with a backup list and a way to query which nodes/orders use a specific location then have an option to use the backup location or switch off entirely; finally implement a way to switch the whole game off based on these queries.

**Comment by Jacki3** [comment]
One can use the `IsWithinRadius` method in `SpawnOnMap` which can determine if a location (a string of lat/lon) is within a radius given the centre of that circle and the radius in meters. We then need to build backup locations which will be stored in Flow Engine that will be used if the given location is within the radius. Otherwise, nodes/orders or games can be shut off if location is within radius without a backup; need to hook this into web services and build web app for this work nicely.

Using a custom API should allow other devs who do not use LUTE exclusively to do the same if they wish.

**Comment by Jacki3** [comment]
Need to build a 'LocationBackup' variable which uses a a location and another unique location (or list of locations) which will then be given to our web API that returns a usable location or not - Unity then handles logic based on this return.

**Comment by Jacki3** [comment]
Implemented the LocationBackup variable and editor.

The issue with the editor and using a custom property drawer is that the list of locations to be built looks a bit untidy in this fashion. It will work but the ideal solution might be to create this list seperately (on another object which can be a prefab) so this can be duplicated across locations without recreating the same list (I suppose this is 'first class' location lists).

**Comment by Jacki3** [comment]
Implementing an offline version of this where players will be able to inform the system they cannot access a specific location. The concept of the `LocationFailureHandler` is to store a list of potential locations that will not be accesible with a list of methods to call if the location is called inaccesible.

The location handle failure is a list of `FailureMethods` which uses the queried location, a series of methods to call in the scenario where that location is inaccesible as well as various properties that will be used in specific methods (such as backup locations).

The core system is in place but the editor side needs improving alongside implementing the remaining methods.

**Comment by Jacki3** [comment]
Implemented three methods:
- Backup Node: will execute another node if a specific location cannot be found
- Disable location behaviours: disables all nodes/orders that utilise a specific location
- Nearest Location: will search for all locations and replace the inaccessible location with the nearest to it

TODO:
- Implement rest of methods
- Update editor visuals and logic
- Automatically setup all locations to use this system with one button

**Comment by Jacki3** [comment]
This feature is now implemented besides one feature:

- Allowing players to assign specific locations that are inaccesible rather than automatically choosing

**Comment by Jacki3** [comment]
Allowing players to assign specific locations that are inaccessible rather than automatically choosing using the additional menu options feature on the location failure handler menu

- The latest commit now closes this issue and will be merged into the main branch
